aten/src/ATen/cuda/CUDAGraphsUtils.cuh

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <ATen/CUDAGeneratorImpl.h>
+#include <ATen/cuda/CUDAEvent.h>
+#include <c10/core/StreamGuard.h>
+#include <c10/cuda/CUDAGuard.h>
+
+namespace at {
+namespace cuda {
+namespace philox {
+
+// We can't write a __device__ function in CUDAGeneratorImpl.h, because it's in ATen.
+// Also, whatever call unpacks PhiloxCudaState in consumer kernels must be inlineable.
+// Easiest thing that comes to mind is, define a free function here, in ATen/cuda.
+// Any cuda consumer can include this header.
+__device__ __forceinline__ std::tuple<uint64_t, uint64_t>
+unpack(at::PhiloxCudaState arg) {
+  if (arg.captured_) {
+    return std::make_tuple(arg.seed_, *(arg.offset_.ptr) + arg.offset_intragraph_);
+  } else {
+    return std::make_tuple(arg.seed_, arg.offset_.val);
+  }
+}
+
+} // namespace philox
+
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
+// Protects against enum cudaStreamCaptureStatus implementation changes.
+// Some compilers seem not to like static_assert without the messages.
+static_assert(int(cudaStreamCaptureStatus::cudaStreamCaptureStatusNone) == 0,
```

Comment:
you no longer need these asserts, do you? Your enum below will use whatever values are in the current implementation

Reply:
I'd still like assurance what the converted values are, in case they change their enum to some exotic type.

Reply:
You mean, the type that's not convertible to int? Because that's the only thing that you rely on below, and compilation of `CaptureStatus` will break if that happens, so I don't see what you are gaining by the asserts here.

Reply:
i'm worried about one possibility:  some enum type with multiple values that convert to the same int value.  it's farfetched...if you still think I should remove the asserts i will.